packages/server/lib/plugins/child/run_require_async_child.js

```diff
@@ -4,6 +4,7 @@ const debug = require('debug')(`cypress:lifecycle:child:run_require_async_child:
 const tsNodeUtil = require('./ts_node')
 const util = require('../util')
 const { RunPlugins } = require('./run_plugins')
+const { bundleRequire } = require('bundle-require')
```

Comment:
Should we move this to be lazily required in the `catch` block? 

Especially since this has a peer-dep of `esbuild`, I'm assuming we're the user is expected to supply that in their project rather than include it in the binary.

If that's the case, do we need an error indicating that they need to install it if we fall through `bundle-require` and we can't `require.resolve` it?

Reply:
I rejiggered this a bit. Dynamically loading bundle-require if the user doesn't have esbuild and then falling back to a non-esbuild node `import`.

**Order of operations is**

1. Try synchronous require (commonjs, typescript w/ ts-node + commonjs via `require(configFile)`)
  1. If synchronous require fails with the ESM error message, Continue
  2. Else, throw
2. Try using esbuild if it exists via `require.resolve('esbuild')`
  1. If the user does not have esbuild, Continue
  2. Else, run config file with it. Throw any errors
3. Try node's real esm loading logic via `await import(configFile)`